core/sql/exp/ExpLOBaccess.cpp

```diff
@@ -1970,7 +1970,14 @@ Ex_Lob_Error ExLob::readCursorData(char *tgt, Int64 tgtSize, cursor_t &cursor, I
               continue;
          }
       }
-      
+      if (operLen == tgtSize)
+        {
+          //The cursor still has data available but wait until redrive since
+          //we have fetched all the data for this extract operation
```

Comment:
The cursor on the full lob data indicates that there is data left. But the extract operation can retrieve justt a portion of the total. The state is maintained in the cursor and will get updated with each "execute" of the extract statement issued by caller of the extract statement.  Agree this comment is confusing. I can just say "return the  data to caller".